math/pronicnumber.go

```diff
@@ -0,0 +1,26 @@
+// pronicnumber.go
+// description: Returns true if the number is pronic, false otherwise
+// details: A number n is a pronic number if
+// it is equal to product of two consecutive numbers m and m+1.
+// wikipedia: https://en.wikipedia.org/wiki/Pronic_number
+// author: Akshay Dubey (https://github.com/itsAkshayDubey)
+// see pronicnumber_test.go
+
+package math
+
+// This function returns true if argument passed to the function is pronic, false otherwise.
+func PronicNumber(n int) (bool, error) {
```

Comment:
You could simplify your code a lot by changing the argument type to `uint` - then you do not need to check if `n` is nonnegative and you do not need to worry about the `error`.

Moreover you can optimize the computation:
- every pronic number is even, so if `n` is odd you can return `false`,
- it is enough to check if `n == x*(x+1)`, where `x` is `floor(sqrt(n))` (pseudo code) - this will reduce the time compexity.

Reply:
`int` is more commonly used than `uint` in code doing actual maths(which will be concerned with whether a number is pronic or not), so I don't think that's a good idea.

Reply:
In this case I would simply return `false` for negative inputs and get rid of the `error` as you suggest [below](https://github.com/TheAlgorithms/Go/pull/595#discussion_r1006852041).

Reply:
All the suggested changes have been pushed, kindly have a look.

Thanks, @raklaptudirm & @vil02 for your suggestions.